### DISCUSSION TITLE
Do loader and/or beforeLoad functions have access to the request object in an SSR context?

### DISCUSSION THREAD

**Discussion by rhyek** [TanStack/router#2287] [General]
Interested in getting access to cookies and headers.

**Comment by drew-dulgar** [comment]
This is something I am struggling with as well. 

On the client side (and using fetch), it's simple enough to use `credentials: 'include` and everything is forwarding as expected. 

However, on the server side I am really struggling to figure out how to include cookies in the request.

It's available as part of the req object on the initial server request, which then calls the StartServer component. But I'm not really sure how to make the cookie available for all of the requests that useQuery preferches. 

Any further progress yourself?
